src/ferui-components/tree-view/tree-node-component.ts

```diff
@@ -0,0 +1,233 @@
+import {
+  Component,
+  Input,
+  OnInit,
+  Output,
+  EventEmitter,
+  TemplateRef,
+  HostBinding,
+  ElementRef,
+  ViewChild,
+} from '@angular/core';
+import {
+  TreeViewEvent,
+  TreeViewEventType,
+  TreeNode,
+  TreeViewColorTheme,
+  TreeNodeDataRetriever,
+  PagedTreeNodeDataRetriever,
+} from './interfaces';
+
+@Component({
+  selector: 'fui-tree-node',
+  template: `
```

Comment:
Please use changeDetection: ChangeDetectionStrategy.OnPush and run the changes manually through `changeDetector.markForCheck();`.

Reply:
OnPush detection added